### PR TITLE
Moving From YAJL to Jansson

### PR DESCRIPTION
**Pull request by xw19**
**Moving YAJL to Jansson**

 Added a temporary function to yajl_val to json_t
 converted YAJL_GET_STRING to json_string_value
 converted YAJL_IS_NUMBER to json_is_number
 added functions from double to various int types.
 YAJL_GET_NUMBER returns string. While json_number_value returns double.

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
@giuseppe PTAL

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by giuseppe** [comment]
@saschagrunert are you ok to have this change in the `switch-to-jansson` branch?

**Comment by giuseppe** [comment]
tests are failing

**Comment by saschagrunert** [comment]
@xw19 can you please fixup the tests? :)

**Comment by xw19** [comment]
@giuseppe @saschagrunert  can we retrigger these test builds?

**Comment by xw19** [comment]
I think the issue is caused by arm32v7 latest ubuntu images release last week https://hub.docker.com/r/arm32v7/ubuntu/tags

There is update on the uraimo/run-on-arch-action tag v2.8.1

https://github.com/uraimo/run-on-arch-action/commit/96c27aa6978f736220aea94414920135028189be

Which coincides with the exact line failure 

https://github.com/containers/libocispec/actions/runs/11534806854/job/32210560862

**Comment by giuseppe** [comment]
it seems there is a missing file:

```
  ../../jansson/src/jansson.h:15:10: fatal error: jansson_config.h: No such file or directory
     15 | #include "jansson_config.h"
```

**Comment by xw19** [comment]
I found the cause of the failure was probably because we were not running configure on the jansson hence the jansson_config.h is not being generated.

**Comment by xw19** [comment]
@saschagrunert @giuseppe Can we retry one more time please.
sorry for the inconvenience.

**Comment by xw19** [comment]
WIP: I will first run in my fork and fix then will revert back

**Comment by xw19** [comment]
@saschagrunert @giuseppe Please re-review I think I have fixed the CI issues

https://github.com/xw19/libocispec/actions/runs/11688628819/job/32549520172?pr=1

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -41,10 +41,21 @@ jobs:
             apt-get install -q -y python3 automake libtool autotools-dev git make cmake pkg-config gcc wget xz-utils
 
           run: |
+            # Install jansson
+            cd jansson;
+            autoreconf -fi;
+            ./configure;
+            make;
+            make install;
+            cd ..
```
[the remark this answers is not in the document]
Comment (xw19) [reply .github/workflows/test.yaml:50]: > Can we just install `libjansson-dev` in the above statement?

On ubuntu:2204 jansson is at 2.13 same as Fedora

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by saschagrunert** [comment]
Refers to #138

**Pull request merged**